lib/mayaUsd/listeners/notice.cpp

```diff
@@ -29,28 +29,32 @@ namespace {
 // to be removed.
 static int _newOrOpenRegistrationCount = 0;
 
-static void _OnMayaNewOrOpenSceneCallback(void* /*clientData*/)
+static void _afterNewSceneCallback(void* /*clientData*/) { UsdMayaSceneResetNotice().Send(); }
+
+static void _beforeNewSceneCallback(void* /*clientData*/)
 {
     // kBeforeFileRead messages are emitted when importing/referencing files,
     // which we don't consider a "scene reset".
     if (MFileIO::isImportingFile() || MFileIO::isReferencingFile()) {
         return;
     }
 
+    UsdMayaSceneBeforeResetNotice().Send();
```

Comment:
I'm a bit puzzled by this notification.  Unless I'm mistaken, this is the only place where it's sent, and it's followed immediately by a UsdMayaSceneResetNotice notification.  However, in OpUndoItemList.cpp both notifications do exactly the same thing, which seems redundant: you'll just call OpUndoItemList::instance().clear() twice.  Am I missing something?

Reply:
No, but I did not want to change old user of the previously existing notification and wanted the new notification to cover all "before"cases. As you mention, there will be a 2nd call, but the implementation will clear a variable that is already clear, so the cost is minimal.

I did not want to tie the new "Before" notification to our specific use case and thought the double call innocuous.